src/pages/FoodItem/FoodItemPage.tsx

```diff
@@ -34,7 +34,30 @@ export default function FoodItemPage() {
     document.title = foodItem ? foodItem.name : "Not Found";
   });
 
-  const [totalPrice, setTotalPrice] = useState(foodItem.basePrice);
+  //Create price state. Updates on every input change.
+  const initialPrices = {
+    optionPrices: foodItem.options.map((option) => 0),
+    totalPrice: foodItem.basePrice,
+  };
+  const priceReducer = (
+    state: {
+      optionPrices: number[];
+      totalPrice: number;
+    },
+    action: {
+      index: number;
+      amount: number;
+    }
+  ) => {
+    const newState = { ...state };
+    newState.optionPrices[action.index] = action.amount;
+    newState.totalPrice = newState.optionPrices.reduce(
+      (prev, optionPrice) => prev + optionPrice,
+      foodItem.basePrice
+    );
+    return newState;
+  };
+  const [priceState, dispatchPrice] = useReducer(priceReducer, initialPrices);
```

Comment:
Using reducer instead to manage priceState of each foodItem. Quantity state is managed separately.
Dispatch is then called inside subOptions, whenever there is a change. 

Previous method gets formState and compares with foodItem object, then compares strings to get the price. Very messy way of getting final price.

Reply:
reducers should be defined outside the functional component, it should be a pure function (i.e. not read or modify external data)